rift/node.py

```diff
@@ -1781,6 +1782,11 @@ def store_tie_packet_info(self, tie_packet_info):
             self.peer_node_tie_packet_infos[tie_packet.header.tieid] = tie_packet_info
             self.update_partially_conn_all_intfs()
             self.regenerate_my_south_prefix_tie()
+        if tie_packet.header.level == self.level_value + 1 and \
```

Comment:
I don't think that we can assume that the difference in level between a node and it's parent node is exactly one (i.e. I think that it is possible that level(parent) can be < level(child) + 1).

Certainly there is a lot of language in the draft that appears to imply that the difference is level is exactly one: level(child) == level(parent) + 1.

But there is also some language that in the case of leafs it is only required that level(child) < level(parent), for example:


 4. PROCESS_LIE:
 [...]
 PUSH PODMismatch, PUSH MTUMismatch else
 4. if lie has undefined level OR my level is undefined OR this
 node is leaf and remote level lower than HAT OR (lie’s level
 is not leaf AND its difference is more than one from my
 level) then CLEANUP, PUSH UpdateZTPOffer, PUSH
 UnacceptableHeader else

We need a different test check whether a TIE was received from the parent.